dist/angular-validation.js

```diff
@@ -270,6 +270,28 @@
     };
 
     /**
+     * Add Message Element at the initial phase
+     * Do this function when to custom your messageElement 
+     * NODE: this funtion have similar purpose with `message-id` attribute. 
+     * This function will help you add your `messageElement` automatically instead of pre-defined.
+     * @param element
+     */
+    this.addMsgElement = function(element) {
+      return element.after('<span></span>');
```

Comment:
In my opinion set vs add is absolutely different.
set : `set` value to an existence item/property
add: `add` a not-existence item 
Because Msg Element hadn't existed before I added it => So I use `add` instead of `set`. 
But `set` can be right at some point.

So, I think It's better to merge `addMsgElement` and `setMsgElement` into one `setMsgElement`

In `setMsgElement` we do `add` and return the DOM object of MsgElement. And your directive will get Msg element from returned value from  `setMsgElement`.

**Example:**
````javascript
        $validationProvider.addMsgElement = function(element) {
            $(element).parent().append('<span></span>');
        };
        $validationProvider.getMsgElement = function(element) {
          return $(element).parent().children().last();
        };
````
will be replaced by

````javascript
        $validationProvider.setMsgElement = function(element) {
          $(element).parent().append('<span></span>');
          return $(element).parent().children().last();
        };
````

If you agree with this implement I will update my code.